### PR TITLE
refactor(Scripts/TempleOfAhnQiraj): Clean up Cthun script

### PR DESCRIPTION
**Pull request by Nyeriah**
## Changes Proposed:

This PR proposes changes to:
-  [ ] Core (units, players, creatures, game systems).
-  [x] Scripts (bosses, spell scripts, creature scripts).
-  [ ] Database (SAI, creatures, etc).

## Issues Addressed:

- Closes 

## SOURCE:

The changes have been validated through:
- [ ] Live research (checked on live servers, e.g Classic WotLK, Retail, etc.)
- [ ] Sniffs (remember to share them with the open source community!)
- [ ] Video evidence, knowledge databases or other public sources (e.g forums, Wowhead, etc.)
- [ ] The changes promoted by this pull request come partially or entirely from another project (cherry-pick). **Cherry-picks must be committed using the proper --author tag in order to be accepted, thus crediting the original authors, unless otherwise unable to be found**

## Tests Performed:

This PR has been:
- [x] Tested in-game by the author.
- [ ] Tested in-game by other community members/someone else other than the author/has been live on production servers.
- [ ] This pull request requires further testing and may have edge cases to be tested.

## How to Test the Changes:

- [ ] This pull request can be tested by following the reproduction steps provided in the linked issue
- [ ] This pull request requires further testing. Provide steps to test your changes. If it requires any specific setup e.g multiple players please specify it as well.

1.
2.
3.

## Known Issues and TODO List:

- [ ]
- [ ]

## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.

### CODE REVIEW

File: src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp [hunk 1]
```diff
@@ -448,33 +419,21 @@ struct boss_cthun : public BossAI
         }).Schedule(30s, [this](TaskContext context)
         {
             if (Creature* eye = instance->GetCreature(DATA_EYE_OF_CTHUN))
-            {
                 eye->AI()->DoAction(ACTION_SPAWN_EYE_TENTACLES);
-            }
 
-            context.Repeat(30s);
+            context.Repeat();
```
Comment (sudlud) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp:424]: why did you drop the 30s?

File: src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp [hunk 2]
```diff
@@ -259,9 +235,7 @@ struct boss_eye_of_cthun : public BossAI
                 if (Unit* target = SelectTarget(SelectTargetMethod::Random, 0, 0.0f, true))
                 {
```
Comment (sudlud) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp:236]: these braces can be dropped, too

File: src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp [hunk 3]
```diff
@@ -559,10 +503,8 @@ struct boss_cthun : public BossAI
                     me->RemoveAurasDueToSpell(SPELL_PURPLE_COLORATION);
                     DoCastSelf(SPELL_CARAPACE_CTHUN, true);
                     //Spawn flesh tentacle
-                    for (uint8 i = 0; i < 2; i++)
-                    {
-                        me->SummonCreature(NPC_FLESH_TENTACLE, FleshTentaclePos[i], TEMPSUMMON_CORPSE_TIMED_DESPAWN, 5000);
-                    }
+                    for (const auto& position : FleshTentaclePos)
```
Comment (sudlud) [inline src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp:506]: here you use `const auto`, further up it was `auto const`

File: src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp (hunk 1 above)
Comment (Nyeriah) [reply src/server/scripts/Kalimdor/TempleOfAhnQiraj/boss_cthun.cpp:424]: Because it is 30s in the first call so it’s only necessary to repeat it

**Review by sudlud — APPROVED** [review]

**Pull request merged**